File: adapter/python/custom_serializer_adapter.py
from json_serializer import JsonSerializer
from custom_serializer import CustomSerializer


class CustomSerializerAdapter(JsonSerializer):
    """
    Daha sonradan dahil edilen yapının sisteme adepte edilmesi işlemi.
    Sistemin kullandığı arayüzü uygular.
    Bu sayede IJsonSerializer'i uygulayan bir sınıf örneği istendiğinde
    bu sınıf örneğini vererek mevcut koda dokunmadan sistemin devam etmesi
    sağlanır.
    UML diyagramındaki Adapter'a denk gelmektedir.
    """

    def serialize_object(self, obj):
        custom_serializer = CustomSerializer()
        return custom_serializer.serialize(obj)

File: adapter/go/custom-serializer-adapter.go
package main

// customSerializerAdapter Daha sonradan dahil edilen yapının sisteme adepte edilmesi işlemi.
// Sistemin kullandığı arayüzü uygular.
// Bu sayede JSONSerializer'i uygulayan bir struct örneği istendiğinde
// bu struct örneğini vererek mevcut koda dokunmadan sistemin devam etmesi
// sağlanır.
// UML diyagramındaki Adapter'a denk gelmektedir.
type customSerializerAdapter struct{}

func NewCustomSerializerAdapter() *customSerializerAdapter {
	return &customSerializerAdapter{}
}

func (c *customSerializerAdapter) serializeObject(object []byte) string {
	customSerializer := NewCustomSerializer()
	return customSerializer.Serialize(object)
}
